Add Kubectl example (#237)
tektoncd/catalog: kubectl/README.md

> # Knative with knctl

Reviewer: We should rename it to something like `kubectl` or `Deploy Kubernetes Resource with kubectl`?

Author: Ah, you catch it, my bad.

> # Knative with knctl

Reviewer 2: "Knative"? ...copy-paste artifact? should this say "Kubectl" instead?

Author: Ah, again

This Task deploys (or delete) a Kubernates resource (pod). It uses
[`kubectl`](https://kubernetes.io/zh/docs/reference/kubectl/kubectl/) for that.

## Install the Task

```
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/kubectl/kubectl-deploy.yaml
```

## Install ClusterRolebinding

```
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/kubectl/clusterrolebinding.yaml
```

## Inputs 

### Parameters

> * **resource:**: The content of the resource to deploy

Reviewer 2: could this be called `manifest` to make it clear that this is not a name or reference but a configuration kind of YAML string?

also, in order to use this in the [KFP-Tekton compiler](https://github.com/kubeflow/kfp-tekton/issues/49) perspective, we would need the following parameters:

- `action`: `['get', 'create', 'apply', 'delete', 'replace', 'patch']` -- the action to perform to the resource
- `merge_strategy`: `['strategic', 'merge', 'json']` -- the strategy used to merge a patch, defaults to "strategic"
- `success_condition`: a label selector expression which describes the `success` condition
- `failure_condition`: a label selector expression which describes the `failure` condition
- `manifest`: the kubernetes manifest

Even though not all of them will be supported yet, can we start out with at least `action` and `manifest` and document that there is a limitation for `action` for now?

I am worried that once this is getting used by a larger group of consumers, it will be hard to make substantial changes to parameters without breaking downstream pipelines

Author: Sorry for delay, involved in a tough project.
The function you mentioned is correct and needed for a complete `kubernete resource hendler` features, it's too complex to just with `script` and `yml`, I think it should be a pre-prepared image.

Whatever, I will dig into and check if i can found something.

## Usage

> This TaskRun runs the Task to deploy the given image as a Knative service.

Reviewer: ```suggestion
This TaskRun runs the Task to deploy the given Kubernetes resource definition.
```

Author: Done, thanks

> This TaskRun runs the Task to deploy the given image as a Knative service.

Reviewer 2: is this generic enough to support any K8s resource similar to what Argo supports with `ResourceTemplate` allowing actions like 'get', 'create', 'apply', 'delete', 'replace', 'patch'?

Author: I think it could, although it's simple, for example, for `patch`, you could define one more parameter named `resource-name`, and use `kubectl patch $(resource-name) -f `.
for `get`, you could echo the result to a volume and it could be checked by next steps.
